src/apps/shared/test/client.py

```diff
@@ -9,31 +11,85 @@ class TestClient:
     def __init__(self):
         app = create_app()
         self.client = AsyncClient(app=app, base_url="http://test.com")
+        self.headers = dict()
 
     @staticmethod
     def _prepare_url(url, query):
         return f"{url}{urllib.parse.urlencode(query)}"
 
-    async def post(self, url, query=None, data=None, headers=None) -> Response:
+    def _get_headers(self, headers: Optional[dict] = None) -> dict:
+        headers_ = dict(self.headers)
+        if headers:
+            headers_.update(headers)
+        return headers_
+
+    @staticmethod
+    def _get_body(data: Optional[dict] = None):
+        if data:
+            return json.dumps(data)
+        return {}
+
+    async def post(
+        self,
+        url: str,
+        data: Optional[dict] = None,
+        query: Optional[dict] = None,
+        headers: Optional[dict] = None,
+    ) -> Response:
         if query:
             url = self._prepare_url(url, query)
-        response = await self.client.post(url, data=data, headers=headers)
+        response = await self.client.post(
+            url, data=self._get_body(data), headers=self._get_headers(headers)
+        )
         return response
 
-    async def put(self, url, query=None, data=None, headers=None):
+    async def put(
+        self,
+        url: str,
+        data: Optional[dict] = None,
+        query: Optional[dict] = None,
+        headers: Optional[dict] = None,
+    ) -> Response:
```

Comment:
We can use new python 3.10 format for type annotations

```python
dict | None
```

Reply:
Also I don't think that put request could have data that is `None`

Reply:
Done